packages/wrangler/src/deployment-bundle/node-compat.ts

```diff
@@ -17,85 +16,76 @@ export type NodeJSCompatMode = "legacy" | "v1" | "v2" | null;
  * Returns one of:
  *  - "legacy": build-time polyfills, from `node_compat` flag
  *  - "v1": nodejs_compat compatibility flag
- *  - "v2": experimental nodejs_compat_v2 flag
+ *  - "v2": nodejs_compat_v2 compatibility flag
  *  - null: no Node.js compatibility
  *
- * Currently we require that the v2 mode is configured via `experimental:nodejs_compat_v2` compat flag,
- * where the `experimental:` prefix is stripped before being passed to the runtime since that does not
- * understand this prefix.
+ * Currently v2 mode is configured via `nodejs_compat_v2` compat flag.
+ * At a future compatibility date, the use of `nodejs_compat` flag will imply `nodejs_compat_v2`.
  *
  * We assert that only one of these modes can be specified at a time.
  * We assert that you must prefix v2 mode with `experimental`.
  * We warn if using legacy or v2 mode.
  */
-export function validateNodeCompat({
-	legacyNodeCompat,
-	compatibilityFlags,
-	noBundle,
-}: {
-	legacyNodeCompat: boolean;
-	/* mutate */ compatibilityFlags: string[];
-	noBundle: boolean;
-}): NodeJSCompatMode {
-	if (legacyNodeCompat) {
-		logger.warn(
-			"Enabling Wrangler compile-time Node.js compatibility polyfill mode for builtins and globals. This is experimental and has serious tradeoffs."
+export function validateNodeCompat(
+	compatibilityFlags: string[],
+	nodeCompat: boolean | undefined,
+	noBundle: boolean | undefined
+): NodeJSCompatMode {
+	const {
+		mode,
+		nodejsCompat,
+		nodejsCompatV2,
+		experimentalNodejsCompatV2,
+		legacy,
+	} = getNodeCompatMode(compatibilityFlags, nodeCompat);
+
+	if (experimentalNodejsCompatV2) {
```

Comment:
Maybe throw inside of `getNodeCompatMode`so that there is no need to return `experimentalNodejsCompatV2`